admin/release

```diff
@@ -0,0 +1,162 @@
+#~/bin/bash
+
+## Build and deploy framework release
+
+UPSTREAM=https://github.com/bcit-ci/CodeIgniter4.git
+action=test
+version=4
+qualifier=
+
+branch=release-
+devonly='.github/* admin/* build/* contributing/* user_guide_src/* CODE_OF_CONDUCT.md \
+DCO.txt PULL_REQUEST_TEMPLATE.md'
+which=release
+
+BOLD='\033[1m'
+NORMAL='\033[0m'
+COLOR='\033[1;31m'
+ERROR='\033[0;31m'
+
+echo -e "${BOLD}${COLOR}CodeIgniter4 release builder${NORMAL}"
+echo '----------------------------'
+
+#---------------------------------------------------
+# Check arguments
+echo -e "${BOLD}Checking arguments...${NORMAL}"
+
+if [ $# -lt 1 ]; then
+    echo "You really need to read the directions first!"
+    exit 1
+fi
+
+if [ $1 = 'deploy' ]; then
+    action=deploy
+elif [ $1 != 'test' ]; then
+    echo -e "${ERROR}Invalid action ($1)${NORMAL}"
+    exit 1
+fi
+shift   # drop the command from the argument list
+
+version=$1
+if [ $# -gt 1 ]; then
+    qualifier="-${2}"
+    which='pre-release'
+fi
+branch="release-$version$qualifier"
+
+#---------------------------------------------------
+# Create the release branch
+echo -e "${BOLD}Creating $which $branch to $action ${NORMAL}"
+
+git checkout develop
+git branch -d $branch     # remove the branch if there
+git checkout -b $branch
+
+#---------------------------------------------------
+# Update version dependencies
+echo -e "${BOLD}Updating version dependencies${NORMAL}"
+
+# CI_VERSION definition in system/CodeIgniter.php
+sed -i "/const CI_VERSION/s/'.*'/'${version}${qualifier}'/" system/CodeIgniter.php
+
+# release substitution variable in user_guide_src/source/conf.py
+sed -i "/release =/s/'.*'/'${version}${qualifier}'/" user_guide_src/source/conf.py
+
+# version & date in user_guide_src/source/changelog.rst
+sed -i "/|release|/s/|.*|/${version}${qualifier}/" user_guide_src/source/changelog.rst
```

Comment:
I haven't really checked, but his looks like it might modify older version numbers.

Reply:
I edited checked the conf.py, CodeIgniter.php and changelog.rst files, if needed, to make sure there was a unique pattern I could search for. In the case of conf,py and CodeIgniter.php, it was the version's constant declaraiton (only there once). In the case of the changelog,rst, I modified it so that the upcoming release block was driven by the sphinx substitution.
The "fallback" provision is that the new release branch is only built locally, and in the local clone.
I will look into beefing it up a bit to check that there is only the one occurrence of each pattern.